### ISSUE TITLE
[show-gpus] Show real-time spot price

### ISSUE THREAD

**Issue by Michaelvll** [skypilot-org/skypilot#1438]
From our user:
> Additionally, can the sky show-gpus command line tool be extended to display real-time spot price too?

**Comment by concretevitamin** [comment]
As a first cut, maybe we can implement a `--refresh` option for `show-gpus`, which calls the catalog fetcher to pull latest prices and save to local files.

**Comment by csaroff** [comment]
What's the current state of this? Do I need to manually refresh spot prices?

**Comment by concretevitamin** [comment]
@csaroff For AWS and GCP, the hosted catalog will refresh every 7 hours ([ref](https://github.com/skypilot-org/skypilot-catalog?tab=readme-ov-file#automatic-catalog-fetching)), and they will be pulled down to local `~/.sky/catalogs`. This means in `sky show-gpus / launch` etc. the spot prices are mostly up to date, just not actually *real-time*. 

We felt this freshness is reasonable since GCP guarantees at most once-per-month spot pricing changes, and AWS spot prices have stabilized a lot too. 

Is this freshness good enough? Note the cloud(s) will bill users using the actual prices.

**Comment by csaroff** [comment]
Yeah, but I'm seeing some inconsistencies:

<img width="1171" alt="image" src="https://github.com/skypilot-org/skypilot/assets/7225723/a54a0901-b82b-47f8-a83c-1781c1e778b1">

<img width="861" alt="image" src="https://github.com/skypilot-org/skypilot/assets/7225723/198fcafe-92a4-4baf-ae9a-522b85ea5201">

https://cloud.google.com/compute/all-pricing#a2-ultra-machine-types

According to the gcp catalog, this instance should cost 2 * $6.4/hour=$12.8/hour right?

**Comment by concretevitamin** [comment]
GCP seems to have some contradictions.

**The actual console** shows the following for `a2-ultragpu-4g`, spot, in `us-central1`:

<img width="388" alt="Screen Shot 2023-12-13 at 21 00 31" src="https://github.com/skypilot-org/skypilot/assets/592670/cfa8228a-773c-4f34-add7-9bce037690a3">

This $8.17 number is close to the price fetched by our catalog, and displayed in the optimizer table above ($8.04 / hour / node).

**The linked [page](https://cloud.google.com/compute/all-pricing#a2-ultra-machine-types)** shows $6.4 and claims 
> This total cost includes the cost for the GPUs, vCPU, memory, and local SSD.

This statement, or the $6.4 number, may be outdated.

tldr: SkyPilot's catalog & optimizer show the same price as the GCP console :)

**Comment by csaroff** [comment]
Got it! Really appreciate the clarification
